### PR TITLE
Add purchaseData parameter for pay quote APIs

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [ ] Internal API changes: explain the internal logic changes

## How to test

- [ ] Automated tests: link to unit test file
- [ ] Manual tests: step by step instructions on how to test

## Contributor NFT

Paste in your wallet address below and we will airdrop you a special NFT when your pull request is merged. 

```Address: ```

---

## PR-Codex overview
The focus of this PR is to add a `purchaseData` parameter to store extra details for purchases in various functions and UI components.

### Detailed summary
- Added `purchaseData` parameter in functions and UI components for storing extra purchase details
- Required `fromAddress` parameter added in `getBuyWithFiatQuote`
- Updated multiple functions to include `purchaseData` for purchase details storage

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by joaquim-verges — APPROVED** [review]

File: packages/thirdweb/src/pay/buyWithFiat/getQuote.ts [hunk 1]
```diff
@@ -236,33 +243,26 @@ export async function getBuyWithFiatQuote(
...
-
-    const response = await getClientFetch(params.client)(url);
+    const response = await clientFetch(getPayBuyWithFiatQuoteEndpoint(), {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        toAddress: params.toAddress,
```
Comment (IDubuque) [inline packages/thirdweb/src/pay/buyWithFiat/getQuote.ts:255]: can we also add fromAddress in here? It will be optional field

**Review by IDubuque — APPROVED** [review]
lgtm

**Pull request merged**